app/src/main/java/com/android/voyageur/ui/gallery/PermissionButtonForGallery.kt

```diff
@@ -0,0 +1,144 @@
+package com.android.voyageur.ui.gallery
+
+import android.Manifest.permission.READ_EXTERNAL_STORAGE
+import android.Manifest.permission.READ_MEDIA_IMAGES
+import android.Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.net.Uri
+import android.os.Build
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+
+@Composable
+fun PermissionButtonForGallery(
+    onUriSelected: (Uri?) -> Unit,
+    messageToShow: String,
+    dialogMessage: String,
+    modifier: Modifier = Modifier
+) {
+  val context = LocalContext.current
+  var showRationaleDialog by remember { mutableStateOf(false) }
+  val permissionVersion =
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        READ_MEDIA_IMAGES
+      } else {
+        READ_EXTERNAL_STORAGE
+      }
```

Comment:
For the moment, I will leave this as it is since without a custom photo picker I don't necessarily see why we should ask for `READ_MEDIA_VISUAL_USER_SELECTED` , as we are already using the android default picker which doesn't give access except to what the user selects. In the future however, we could adapt the app to provide additional support for this :)